src/bsk201909/secrets.md

```diff
@@ -0,0 +1,264 @@
+## How do you securely manage secrets in Kubernetes?
+
+Kubernetes secrets hold the most sensitive information of your application - API keys, tokens, database passwords, etc.
+
+If a hacker can retrieve one of these secrets, they could connect to your database without you even noticing it.
+
+It's important to ensure that those secrets are stored as securely as possible.
+
+Let's recap on how secrets work in Kubernetes.
+
+Secrets are objects that contain key-value pairs, and some metadata.
+
+Secrets are similar to ConfigMaps and share the same limitations (1MB in size — as an example).
+
+The main differences are the security protections added by Kubernetes teams to secrets.
+
+This is how a secret looks like in a YAML representation:
+
+```yaml|title=secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: MWYyZDFlMmU2N2Rm
+```
+
+You can create a secret like any other Kubernetes object (usually with kubectl).
+
+The manifest will be sent to the server that will proceed it and store it in etcd - similarly to all Kubernetes objects.
+
+_So why did the Kubernetes team decided to “encrypt” the secrets using base64?_
+
+As you might already know, [base64 is an encoding, not an encryption](https://en.wikipedia.org/wiki/Base64).
+
+Encoding allows you to represent binary data in a manifest.
+
+Imagine storing a certificate in Kubernetes without base64: a lot of "���" — unrecognised characters.
+
+Base64 translates those binaries files in standard strings such as "aGVsbG8gdGhlcmUh==".
+
+But we still haven't answered how to properly secure those secrets.
+
+In Kubernetes, you can opt-in to encryption at rest: by enabling this feature, Kubernetes API encrypts the secrets (optionally, using an external KMS system) before storing them in etcd.
+
+You solved the issue of storing sensitive files such as certificates inside the cluster.
```

Comment:
Not clear that this refers to the encryption at rest feature:

```markdown
So, encryption at rest solves the issue of securing your sensitive data inside the cluster
```

Reply:
Make the sentence clearer